### PR TITLE
improved categorization renderer

### PR DESCRIPTION
**Pull request by eneufeld**
I improved the categorization renderer to support multiple levels of categories. (the current renderer can only renderer the first level).
It also fixes some styling issues.

### CODE REVIEW

File: examples/app/person/person.controller.js [hunk 1]
```diff
@@ -231,17 +231,56 @@ angular.module('makeithappen').controller('PersonController', function() {
...
-                    "effect":"SHOW",
-                    "condition":{
-                        "type":"LEAF" ,
-                        "scope": {
-                            "$ref": "#/properties/personalData/properties/age"
-                        },
-                        "expectedValue":36
-                    }
-                },
+                // "rule":{
```
Comment (edgarmueller) [inline examples/app/person/person.controller.js]: Why is this commented out?

File: src/components/renderers/layouts/categories/categorization.directive.ts [hunk 2]
```diff
@@ -15,30 +15,51 @@ class CategorizationController  extends AbstractLayout {
...
+        clickScope.expanded = !clickScope.expanded;
+        if (!clickScope.expanded) {
+          this.selectedCategory = null;
+        }
+      }
     }
 }
 
-const categorizationTemplate = `<jsonforms-layout>
+const categorizationTemplate = `
```
Comment (edgarmueller) [inline src/components/renderers/layouts/categories/categorization.directive.ts]: Do we also need to update the material and bootstrap renderers? Running this with material looks a bit weird a `Sub` tab appear with not content.

**Comment by eneufeld** [comment]
@edgarmueller Could you take a look again 
- updated the bootstrap and material renderer
- reverted commented example code

**Comment by edgarmueller** [comment]
Looks good, thanks.

**Pull request merged**